tests/common.py

```diff
@@ -1741,14 +1742,17 @@ async def snapshot_platform(
     entity_registry: er.EntityRegistry,
     snapshot: SnapshotAssertion,
     config_entry_id: str,
+    platform: Platform | None = None,
 ) -> None:
     """Snapshot a platform."""
     entity_entries = er.async_entries_for_config_entry(entity_registry, config_entry_id)
     assert entity_entries
-    assert (
+    assert platform or (
```

Comment:
There are quite a few integrations where patching the list/set of platforms is either undesirable or would result in rewriting hundreds of lines of tests.

Reply:
That was exactly my intention for this change 👍

Reply:
I don't think the patching is bad, they way it is done isn't pretty.
Would be nicer if we have a fixture we could just parameterize instead, removing the need for all these patches everywhere.

Such a generic fixture could be beneficial for more tests (and thus speed up integration setup in tests), and serve a wider purpose than just these snapshot helpers.

../Frenck